src/app/application.cpp

```diff
@@ -771,6 +795,64 @@ void Application::shutdownCleanup(QSessionManager &manager)
 }
 #endif
 
+#ifdef Q_OS_WIN
+namespace
+{
+    std::uint64_t autoMemoryWorkingSetSize(const std::uint64_t memorySize)
+    {
+        const std::uint64_t GB = 1024 * 1024 * 1024;
+
+        return (memorySize <= (4 * GB))
+                ? memorySize / 2
+                : (2 * GB) + (memorySize / 10);
+    }
+}
+
+void Application::applyMemoryWorkingSetLimit()
+{
+    MEMORYSTATUSEX statex;
+    statex.dwLength = sizeof(statex);
+    if (!::GlobalMemoryStatusEx(&statex))
+    {
+        const DWORD errorCode = ::GetLastError();
+        QString reason = tr("Error %1.").arg(errorCode);
+        LPVOID lpMsgBuf = nullptr;
+        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+                         , nullptr, ::GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+        {
+            reason = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
+            ::LocalFree(lpMsgBuf);
+        }
+        LogMsg(tr("Failed to set Memory Working Set limit. Reason: %1").arg(reason), Log::WARNING);
+        return;
+    }
+
+    const int MB = 1024 * 1024;
+    const int limit = memoryWorkingSetLimit();
+    const SIZE_T maxSize = ((limit <= 0) ? autoMemoryWorkingSetSize(statex.ullTotalPhys) : (statex.ullTotalPhys * limit / 100));
+    const SIZE_T minSize = (maxSize < (64 * MB)) ? maxSize : (64 * MB);
+    if (::SetProcessWorkingSetSizeEx(::GetCurrentProcess(), minSize, maxSize, QUOTA_LIMITS_HARDWS_MAX_ENABLE))
+    {
+        const QString displayValue = (limit <= 0) ? tr("Auto") : (QString::number(limit) + u'%');
+        LogMsg(tr("Memory Working Set limit is successfully applied. Used value: %1 (%2 MB).")
```

Comment:
Memory working set limit is applied. Value: %1 (%2 MB)

Reply:
On a second thought, operation success messages can be omitted because application is working as intended.

Reply:
> On a second thought, operation success messages can be omitted because application is working as intended.

I also came up with the same idea.